src/UnderlineNav2/UnderlineNav.tsx

```diff
@@ -3,34 +3,32 @@ import Box from '../Box'
 import sx, {merge, BetterSystemStyleObject, SxProp} from '../sx'
 import {UnderlineNavContext} from './UnderlineNavContext'
 import {useResizeObserver, ResizeObserverEntry} from '../hooks/useResizeObserver'
-import CounterLabel from '../CounterLabel'
 import {useTheme} from '../ThemeProvider'
 import {ChildWidthArray, ResponsiveProps, ChildSize} from './types'
 import VisuallyHidden from '../_VisuallyHidden'
 import {moreBtnStyles, getDividerStyle, getNavStyles, ulStyles, menuStyles, menuItemStyles, GAP} from './styles'
 import styled from 'styled-components'
-import {LoadingCounter} from './LoadingCounter'
 import {Button} from '../Button'
 import {TriangleDownIcon} from '@primer/octicons-react'
 import {useOnEscapePress} from '../hooks/useOnEscapePress'
 import {useOnOutsideClick} from '../hooks/useOnOutsideClick'
 import {useId} from '../hooks/useId'
 import {ActionList} from '../ActionList'
 import {defaultSxProp} from '../utils/defaultSxProp'
+import CounterLabel from '../CounterLabel'
+import {LoadingCounter} from './LoadingCounter'
+
+// import {MenuItemLink} from './MenuItemLink'
 
 export type UnderlineNavProps = {
+  children: React.ReactNode
   'aria-label'?: React.AriaAttributes['aria-label']
   as?: React.ElementType
   sx?: SxProp['sx']
-  // cariant and align are currently not in used. Keeping here until some design explorations are finalized.
-  variant?: 'default' | 'small'
```

Comment:
Note for reviewers: These variants were considered in the initial version of the component but it is decided otherwise at the end. Removing them to reduce the clutter and complexity on the component and styles.